trunner/tools/gdb.py

```diff
@@ -151,6 +147,34 @@ def _close(self):
         self.logfile.close()
 
 
+class OpenocdGdbServer(OpenocdProcess):
+    """Handler for OpenocdGdbServer process"""
+
+    @contextmanager
+    @add_output_to_exception(OpenocdError)
+    def run(self):
+        try:
+            # Use pexpect.spawn to run a process as PTY, so it will flush on a new line
```

Comment:
As you can see the whole

```
            # Use pexpect.spawn to run a process as PTY, so it will flush on a new line
            if self.board:
                args = ["-f", f"board/{self.board}.cfg"]
            else:
                args = ["-f", f"interface/{self.interface}.cfg", "-f", f"target/{self.target}.cfg"]

            if self.extra_args:
                args.extend(self.extra_args)

            self.proc = pexpect.spawn("openocd", args, encoding="ascii", logfile=self.logfile)

            try:
                self.proc.expect_exact("Info : Listening on port 3333 for gdb connections")
            except (pexpect.TIMEOUT, pexpect.EOF) as e:
                raise OpenocdError("Failed to start gdb server", self.logfile.getvalue()) from e
```

part is common for both the classes, that's why we can create the additional method, for example called `_start()` which will do this part and then it can be used in:

 - `__call__()` in OpenocdProcess
 - `run()` in `OpenocdGdbServer`

! Important note
Please remember that you have to overwrite the `__call__` method to pass in `OpenocdGdbServer` to avoid executing the code from parent class. For any additional explanations please contact @mateusz-bloch